subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidateTaskPropertiesIntegrationTest.groovy

```diff
@@ -387,6 +391,58 @@ class ValidateTaskPropertiesIntegrationTest extends AbstractIntegrationSpec {
         file("build/reports/task-properties/report.txt").text == ""
     }
 
+    @Unroll
+    def "report setters for mutable type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                private ${type} sources;
+                
+                @InputFiles
+                public ${type} getSources() { return sources; } 
+                public void setSources(${type} sources) { this.sources = sources; } 
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        when:
+        fails "validateTaskProperties"
+
+        then:
+        file("build/reports/task-properties/report.txt").text == """
+            Warning: Type 'MyTask': 'sources' of type '${type.replaceAll("<.+>", "")}' has redundant setter method.
+        """.stripIndent().trim()
+
+        where:
+        type << [ConfigurableFileCollection.name, "${Property.name}<String>", RegularFileProperty.name]
+    }
+
+    @Unroll
+    def "don't report when non-final field doesn't have setters for type #type"() {
```

Comment:
```suggestion
    def "don't report property with mutable type if it doesn't have a setter for type #type"() {
```

Reply:
No matter how it's phrased, the title of this test case is complicated. Another approach could be to change `report setters for property with mutable type #type` above, and define the following properties:

```groovy

                @InputFiles
                public ${type} getMutablePropertyWithSetter() { /* ... */ } 
                public void setMutablePropertyWithSetter(${type} sources) { /* ... */ } 

                // Note that this is not causing a warning
                @InputFiles
                public ${type} getMutablePropertyWithoutSetter() { /* ... */ } 
```

I think this would capture what is and what isn't called out in a more concise and easier to follow way.

Reply:
Done

Reply:
Changed the validation to use sets so that the order of warnings is not relevant